mediapipe/graphs/instantmotiontracking/calculators/sticker_manager_calculator.cc

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include "mediapipe/framework/calculator_framework.h"
+#include "mediapipe/framework/port/ret_check.h"
+#include "mediapipe/graphs/instantmotiontracking/calculators/transformations.h"
+
+namespace mediapipe {
+
+constexpr char kStringTag[] = "STRING";
+constexpr char kAnchorsTag[] = "ANCHORS";
+constexpr char kUserRotationsTag[] = "USER_ROTATIONS";
+constexpr char kUserScalingsTag[] = "USER_SCALINGS";
+constexpr char kRenderDescriptorsTag[] = "RENDER_DATA";
+
+// This calculator takes in the sticker data from the application as a string
+// in a sequence of the following format:
+// "(sticker_id:1,sticker_anchor_x:1.44,sticker_anchor_y:0.0,
+// sticker_rotation:0.0,sticker_scaling:0.0,sticker_render_id:0,
+// should_reset_anchor:true)(sticker_id:2.............."
+//
+// Input:
+//  STRING - String of sticker data in appropriate parse format
+// Output:
+//  ANCHORS - Anchors with initial normalized X,Y coordinates
+//  USER_ROTATIONS - UserRotations with radians of rotation from user
+//  USER_SCALINGS - UserScalings with increment of scaling from user
+//  RENDER_DATA - Descriptors of which objects/animations to render for stickers
+//
+// Example config:
+// node {
+//   calculator: "StickerManagerCalculator"
+//   input_stream: "STRING:sticker_data_string"
```

Comment:
Similar to other calculator, let's add descriptions for the types of these streams, which are optional and required, and for any required ones, add them to the GetContract() checks.